scripts/staging/fm_binclass.dml

```diff
@@ -0,0 +1,50 @@
+#-------------------------------------------------------------
+#
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+# 
+#   http://www.apache.org/licenses/LICENSE-2.0
+# 
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+#
+#-------------------------------------------------------------
+
+/*
+ * Factorization Machines for binary classification.
+ */
+
+# Imports
+source("staging/fm.dml") as fm
+source("nn/layers/sigmoid.dml") as sigmoid
+source("nn/layers/log_loss.dml") as log_loss
+
+train = function(matrix[double] X)
```

Comment:
Need `Y` and probably validation versions, i.e., `X_val` & `Y_val`, the factorization dim `k`, as well as any other training parameters that would be nice to be able to supply, such as the number of `epochs`.